api/middlewares/admin.js

```diff
@@ -0,0 +1,26 @@
+const config = require('config');
+
+/**
+ * check the admin's api key
+ * @param {Object} req - HTTP request
+ * @param {Object} res - HTTP response
+ * @param {function} next - do the following
+ * @returns {Object|function} res or next
+ */
+const checkAdmin = (req, res, next) => {
+  let apikeyadmin = config.get('apikeyadmin');
+  try {
+    apikeyadmin = JSON.parse(apikeyadmin);
+  } catch (err) {
+    return res.status(500).json({ message: 'Internal server error in JSON.parse(apikeyadmin)' });
+  }
```

Comment:
The config doesn't change during server execution, so you could move this code outside of this function. That way it will be executed only once during server startup. You may even let the server crash if the config cannot be parsed.